app/services/manage_braintree_donation.rb

```diff
@@ -44,11 +50,33 @@ def organize_params
     }.merge(@additional_values)
   end
 
+  def transaction
+    return @transaction if @transaction
+
+    # We don't use the `is_subscription` flag here because there are multiple ways that it's possible for
+    # a subscription to be structured. That's more for future information retrieval, not for identifying the
+    # internal structure of the braintree result we're searching through to manage notifications.
+    if @braintree_result.transaction.present?
+      # This is a one-off donation, so we can just use the built in transaction and send the data to the queue.
+      @transaction = @braintree_result.transaction
+    elsif @braintree_result.transactions.present?
+      # This is a "start" subscription event, so we have an array of transactions. We can safely withdraw the most recent
+      # transaction and use that information to send to the queue.
+      @transaction = @braintree_result.transactions.last
```

Comment:
it would be great to include a test in the spec that shows that we want `last` here and not `first`, cause to me if it's the starting transaction in an array of ones that will happen, I would expect it to be `first`